### PR TITLE
Improve usability of the search result autocomplete when typing in via keyboard

### PR DESCRIPTION
**Pull request by cmdcolin**
This is a proposed fix for #2429 

The problem shown in that issue is that: when the user is trying to type in a refName e.g. typing 10<ENTER>, the search results from trix may come in and clobber their attempt at navigating by refName (hitting ENTER would then navigate to whatever is in the trix results)

Try typing e.g. 10 here https://jbrowse.org/code/jb2/v1.5.0/?config=test_data%2Fconfig_demo.json&session=share-Uk4KG8ocZC&password=VXmsd

Functionally, the change on this branch is that if you type something that matches a refname, it will return the matching refnames first in the list (sliced to limit 10 results, arbitrary limit to just balance trix/refname result if they were trying to query trix) and after that in the list, any matching search results from trix.

Internally, we have remove a little bit of duplication of the "fetchResults" function which has appeared in refnameautocomplete,lineargenomeview, and importform. Instead, now we make it so that the user now actually supplies fetchResults as a callback prop, which reduces duplication somewhat and allows a little inversion of control (https://en.wikipedia.org/wiki/Inversion_of_control). 

This PR tries to strike a balance between using the search box for navigating by refnames via keyboard and still allow intuitive searching behavior, with the assumption that in most cases, the what the user types in to navigate to a refname shouldn't collide too much with what they type in to search for a search for a feature

### CODE REVIEW

**Comment by cmdcolin** [comment]
main
https://jbrowse.org/code/jb2/main/?config=test_data%2Fconfig_demo.json&session=share-Uk4KG8ocZC&password=VXmsd

this branch
https://jbrowse.org/code/jb2/refname_autocomplete_always/?config=test_data%2Fconfig_demo.json&session=share-Uk4KG8ocZC&password=VXmsd

**Comment by rbuels** [comment]
can we add the ref seq aliases to the autocomplete as well?

**Comment by rbuels** [comment]
does it make sense to make this reference sequence searching be just a new source of text searching data? Other views are going to have searches, and we are going to have some kind of "global" search as well.

File: plugins/linear-genome-view/src/LinearGenomeView/components/Header.tsx [hunk 1]
```diff
@@ -173,6 +184,7 @@ const LinearGenomeViewHeader = observer(
           <RefNameAutocomplete
             onSelect={handleSelectedRegion}
             assemblyName={assemblyName}
+            fetchResults={fetchResults}
```
Comment (teresam856) [inline plugins/linear-genome-view/src/LinearGenomeView/components/Header.tsx:187]: We moved the fetch results out of the refNameAutocomplete? Is this for more usability in other components?

Comment (cmdcolin) [reply plugins/linear-genome-view/src/LinearGenomeView/components/Header.tsx:187]: yes. this was a architetural idea. it does allow for a little more usability outside lgv this way, or "inversion of control" so the place where it is used can dictate it's how it behaves.

Comment (teresam856) [reply plugins/linear-genome-view/src/LinearGenomeView/components/Header.tsx:187]: okie! good call :) 👍

**Review by teresam856 — APPROVED** [review]
Tested it out, and changes feel good. This is a good addition to this work, maybe for another pr. Just thought I would bring it up here too https://github.com/GMOD/jbrowse-components/issues/2334... This PR looks good to me as is.

**Comment by cmdcolin** [comment]
thanks...definitely agree #2334 would be great

**Pull request merged**